File: Solutions/104. Maximum Depth of Binary Tree/104.java
class Solution {
    public int maxDepth(TreeNode root) {
        if(root==null)return 0;
        if(root.left==null && root.right==null)return 1;
        return Math.max(maxDepth(root.right)+1,maxDepth(root.left)+1);
    }
}

File: Solutions/104. Maximum Depth of Binary Tree/104.py
class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        if root == None:
            return 0
        if root.left == None and root.right == None:
            return 1
        return max(self.maxDepth(root.left)+1, self.maxDepth(root.right)+1)
    